bptestrunner/bluepill_batch_test_runner.template.sh

```diff
@@ -64,8 +64,9 @@ if [ -f "$BP_TEST_ESTIMATE_JSON" ]; then
     TIME_ESTIMATE_ARG="--test-time-estimates-json $(basename "$BP_TEST_ESTIMATE_JSON")"
 fi
 
-# Copy rule-generated test plan file to working folder
-cp "$BP_TEST_PLAN" $BP_WORKING_FOLDER
+# Expand $TEST_UNDECLARED_OUTPUTS_DIR in rule-generated test plan file
+# And copy it to working folder
+sed 's/$TEST_UNDECLARED_OUTPUTS_DIR/'"${TEST_UNDECLARED_OUTPUTS_DIR//\//\\/}"'/g' $BP_TEST_PLAN > $BP_WORKING_FOLDER/$BP_TEST_PLAN
```

Comment:
Should this be conditioned on whether the variable is set? Does it work if you don't have `TEST_UNDECLARED_OUTPUTS_DIR`?

Reply:
If the variable is not set, then the output will be the same. `sed` command won't change the input when the variable is not found. I tested the case when there's no `TEST_UNDECLARED_OUTPUTS_DIR` and verified the content of the test plan json not changed.

Reply:
Oh I got what you meant. `TEST_UNDECLARED_OUTPUTS_DIR` is always set in the `bazel test` world. It's also being used on the [line 84](https://github.com/MobileNativeFoundation/bluepill/blob/master/bptestrunner/bluepill_batch_test_runner.template.sh#LL84C12-L84C12) of the same file.

What I was trying to do is to search the existence of `"$TEST_UNDECLARED_OUTPUTS_DIR"` in the generated test plan and replace it with the actual path, as the test process doesn't know about this env